lib/urlwatch/filters.py

```diff
@@ -809,6 +809,24 @@ def filter(self, data, subfilter):
         return separator.join(sorted(data.split(separator), key=str.casefold, reverse=reverse))
 
 
+class RemoveDuplicateLinesFilter(FilterBase):
+    """Remove duplicate lines"""
+
+    __kind__ = 'remove-duplicate-lines'
+
+    __default_subfilter__ = 'separator'
+
+    def filter(self, data, subfilter):
+        separator = subfilter.get('separator', '\n')
+        data_lines = data.split(separator)
+        uniq_lines = []
+        for line in data_lines:
+            if line not in uniq_lines:
+                uniq_lines.append(line)
```

Comment:
Depending on how many lines there are filtered, it might be faster (at the cost of additional memory) to put the "seen lines" in a `set`, since `set` membership test is cheaper than list membership test.

Reply:
Since there's no ordered set in Python, you'd need to keep the elements around twice, but since the only use for the `uniq_lines` list is to join it, this can be turned into a generator function (inline or otherwise):

```python3
def get_unique_lines(data_lines):
    seen = set()
    for line in data_lines:
        if line not in seen:
            yield line
            seen.add(line)

return separator.join(get_unique_lines(data_lines))
```

Of course, `str.split()` also returns a list of strings, when it could in theory return an iterable, but [since the proposed solutions](https://stackoverflow.com/questions/3862010) are all kind of cumbersome, let's not go that rounte.